Source/WebKit/Platform/IPC/cocoa/SharedFileHandleCocoa.cpp

```diff
@@ -40,20 +44,19 @@ void SharedFileHandle::encode(Encoder& encoder) const
 {
     mach_port_name_t fileport = MACH_PORT_NULL;
     if (fileport_makeport(m_handle.handle(), &fileport) == -1) {
-        encoder << MachPort();
         return;
     }
-
-    encoder << MachPort(fileport, MACH_MSG_TYPE_MOVE_SEND);
+    MachSendRight right = MachSendRight::adopt(fileport);
```

Comment:
`encoder << MachSendRight::adopt(fileport);`

E.g. we say "SharedFileHandle is encoded by creating a send right out of the file handle and encoding the send right".

as opposed of "... is encoded by creating a send right out of the file handle and creating Attachment out of the send right and encoding the Attachment".

Reply:
... If the intention is to prevent the copy, and you've checked that the operator << refuses to invoke the rvalue ref version of the encoding, feel free to ignore this comment